ext/mri/bcrypt_ext.c

```diff
@@ -1,20 +1,57 @@
 #include <ruby.h>
 #include <ow-crypt.h>
 
+#ifdef HAVE_RUBY_THREAD_H
+#include <ruby/thread.h>
+#endif
+
+/* Delete this when 1.8 support is dropped. */
+#ifdef HAVE_RB_STR_NEW_FROZEN
+  #define RB_STR_NEW_FROZEN rb_str_new_frozen
+#else
+  #define RB_STR_NEW_FROZEN rb_str_new4
+#endif
```

Comment:
`HAVE_RUBY_THREAD_H` is in [since 2.0](https://github.com/ruby/ruby/commit/ce0b4fea35c8e692c636ac2cf051ab260e1166f1) and `RB_STR_NEW_FROZEN ` can be dropped for `rb_str_new4 ` according to the comment, now that master is targeting a 4.0 release without Ruby 1.8 or 1.9 support. Was hoping this could get merged, happy to help clean up the patch is so desired.